src/prometheus_alert_rules/memory.rules

```diff
@@ -0,0 +1,69 @@
+groups:
+- name: HostMemory
+  rules:
+  - record: node_memory_MemUsed_percentage
+    expr: |
+      100 * (1 - (
+        (
+          node_memory_MemFree_bytes
+          + node_memory_Cached_bytes
+          + node_memory_Buffers_bytes
+          + node_memory_SReclaimable_bytes
+        ) / node_memory_MemTotal_bytes
+      ))
+  - record: node_memory_SwapUsed_percentage
+    expr: |
+      100 * (1 - (
+        (
+          node_memory_SwapFree_bytes
+          + node_memory_SwapCached_bytes
+        ) / node_memory_SwapTotal_bytes
+      ))
+  - alert: HostMemoryFillsUp
+    expr: predict_linear(node_memory_MemUsed_percentage[30m], 5*60) >= 90
+    for: 2m
+    labels:
+      severity: warning
+    annotations:
+      summary: '[Prediction] Host memory usage will increase to {{ $value | printf "%.0f" }}% in the near future (instance {{ $labels.instance }})'
+      description: >-
+        Host can potentially reach {{ $value | printf "%.0f" }}% memory utilization and risk an OOM kill.
+          VALUE = {{ $value }}
+          LABELS = {{ $labels }}
+        The 5-minute-ahead prediction is made as a linear regression from the last 30 minutes of data.
+  - alert: HostMemoryFull
+    expr: avg_over_time(node_memory_MemUsed_percentage[1m]) > 95
```

Comment:
ditto

Can you also provide some rationale why we alert about reaching 90% soon but this alert threshold is set to 95%?

Reply:
The first alert rule is using linear regression to predict memory usage and raise warning alert rule, in NRPE we had warning and critical thresholds. Instead of warning threshold we are using prediction.
I can add to the first alert condition which will ensure that alert will not fire in permanent memory load. e.g. average usage in last 5min is more than 90 -> prediction will be ignored. This will ensure that the first alert will fire only in trend when memory usage is slowly raising.

Reply:
It makes sense, thanks.

Reply:
I added the `avg_over_time(node_memory_MemUsed_percentage[2m]) < 90` condition to query. Please check tests in description there is example for `HostMemoryFillsUp` alert rule and its behavior during stable high load.